submissions/unabyband/OO-JS/app.js

```diff
@@ -0,0 +1,141 @@
+// Enemies our player must avoid
+// Variables applied to each of our instances go here,
+// we've provided one for you to get started
+// The image/sprite for our enemies, this uses
+// a helper we've provided to easily load images
+const Enemy = function(horizontalMove, verticalPosition, speed) {
+    this.horizontalMove = horizontalMove;
+    this.verticalPosition = verticalPosition;
+    this.speed = speed;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+let winsCounter = 0;
+let speedIncreaser = 100;
+let mainMessage = "";
+let enemy = new Enemy();
+const allEnemies = []; 
+const enemyLocation = [63, 147, 230];
+const enemyStartPosition = -50;
+const enemyEndPosition = 510;
+const enemyWidth = 80;
+const enemyHeight = 60;
+const playerStartHorizontal = 202;
+const playerStartVertical = 405;
+const playerHorizontalStep = 102;
+const playerVerticalStep = 83;
+const maxRounds = 12;
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+// You should multiply any movement by the dt parameter
+// which will ensure the game runs at the same speed for
+// all computers.
+Enemy.prototype.update = function(dt) {
+    this.horizontalMove += this.speed * dt;
+    if (this.horizontalMove > enemyEndPosition) {
+        this.horizontalMove = enemyStartPosition;
+        this.speed = speedIncreaser + Math.floor(Math.random() * 222);
+    };
+    this.checkCollision();
+};
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.horizontalMove, this.verticalPosition);
+};
+
+Enemy.prototype.checkCollision = function () {
+    if (player.horizontalPosition < this.horizontalMove + enemyWidth &&
+        player.horizontalPosition + enemyWidth > this.horizontalMove &&
+        player.verticalPosition < this.verticalPosition + enemyHeight &&
+        player.verticalPosition + enemyHeight > this.verticalPosition) 
+        { 
+        if (winsCounter > 0) {
+            winsCounter -= 1;
+        } else { 
+            winsCounter = 0;
+            speedIncreaser = 100;
+        }
+            player.horizontalPosition = playerStartHorizontal;
+            player.verticalPosition = playerStartVertical;
```

Comment:
Now we are back to the problem with referring the global variable.
The requirements also suggest an apporach on how to resolve the issue.

Reply:
Ok, I've added an argument for tracking player position into Enemy class for checking collisions, hope this is the resolve of the issue (at least the requirements suggest exactly the same, as I understand).
Please, take a look again, thank you.